### ISSUE TITLE
CA1802 false positive on static readonly interpolated strings in Visual Studio 16.9 Preview 3 targeting net472

### ISSUE THREAD

**Issue by KylePolansky** [dotnet/roslyn-analyzers#4732]
🔗 Also filed as [AB#1287517](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1287517)

### Analyzer

**Diagnostic ID**: [CA1802](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1802): `Use Literals Where Appropriate`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

CA1802 is false positively triggering on static readonly strings in Visual Studio 16.9 Preview 3 targeting net472

Similar reported issue: #1175

### Steps To Reproduce

Create a project with the following files in Visual Studio 16.9 Preview 3:

ClassLibrary1.csproj
```
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFramework>net472</TargetFramework>
    </PropertyGroup>
    <ItemGroup>
        <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3" />
    </ItemGroup>
</Project>
```

.editorconfig
```
[*.cs]
dotnet_diagnostic.CA1802.severity = error
```

Class1.cs
```
namespace ClassLibrary1
{
    public class Class1
    {
        private const string foo = "foo";
        private static readonly string fooBar = $"{foo}bar";
    }
}
```

In the above class, CA1802 triggers on line `private static readonly string fooBar = $"{foo}bar";`

However, when changing the line to the suggested `private const string fooBar = $"{foo}bar";`, we get a new error:
```
Error CS8652 : The feature 'constant interpolated strings' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
```

Enabling `preview` language version resolves the `CS8652` as expected, but I would like to use the default language version.

### Expected behavior

CA1802 should not trigger here

### Actual behavior

CA 1802 triggers here

## Additional context

Things tested that do not trigger CA1802 (as expected):
- CA1802 does not appear to be trigger using Visual Studio 16.9 Preview 2 or 16.8.4 (stable)
- CA1802 does not appear to be trigger when targeting `netstandard2.0`, `netcoreapp2.0`, or `net5.0` in the .csproj file

**Comment by Youssef1313** [comment]
@mavasani This would be easy enough to fix here. But I feel like it's a roslyn issue. An `IInterpolatedStringOperation` should not have a `ConstantValue` when the target language version doesn't support constant strings. Currently, the compiler produces constant value for it.

**Comment by mavasani** [comment]
We think the IOperation tree is fine to report a constant value here, but the analyzer should be updated to handle and bail out for this case.

**Comment by MichaCo** [comment]
Just some more information for you, maybe this helps to fix even more issues with this.
I just upgraded to VS 2019 16.9.1 and now getting 100 new CA1802 warnings.

In my solution I only have projects targeting NETSTANDARD 2.0! (not NET4.7.2)

I did try the new package Microsoft.CodeAnalysis.NetAnalyzers, same problem
I did try to update Microsoft.CodeAnalysis.FxCopAnalyzers to 3.3.2, this removed most of the false positives but left a few where I'm using `@$` for example:

```csharp
private static readonly string InsertGroupNodes = $@"
    raiserror('Inserting into {nameof(SomeEntityWhatever)}', 10, 1) with nowait;
```
